plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -77,38 +78,45 @@ mkBuiltinApplication () bn arity0 tys0 args0 =
                 (TypeArg:_,      arg:_,    [])      -> Apply () term arg                        -- type expected, term found
                 _                                   -> term                                     -- something else, including partial application
 
-ckValueToTerm :: () -> CkValue uni fun -> Term TyName Name uni fun ()
-ckValueToTerm () = \case
-    VCon t                           -> t
+ckValueToTerm :: CkValue uni fun -> Term TyName Name uni fun ()
+ckValueToTerm = \case
+    VCon val                         -> Constant () val
     VTyAbs  tn k body                -> TyAbs  () tn k body
     VLamAbs name ty body             -> LamAbs () name ty body
-    VIWrap  ty1 ty2 val              -> IWrap  () ty1 ty2 $ ckValueToTerm () val
-    VBuiltin bn arity0 _ tyargs args -> mkBuiltinApplication () bn arity0 tyargs (fmap (ckValueToTerm ()) args)
+    VIWrap  ty1 ty2 val              -> IWrap  () ty1 ty2 $ ckValueToTerm val
+    VBuiltin bn arity0 _ tyargs args -> mkBuiltinApplication bn arity0 tyargs (fmap ckValueToTerm args)
     {- When we're dealing with VBuiltin we use arity0 to get the type and
        term arguments into the right sequence. -}
 
+newtype CkEnv uni fun = CkEnv
+    { ckEnvRuntime :: BuiltinsRuntime fun (CkValue uni fun)
+    }
+
 instance (Closed uni, GShow uni, uni `Everywhere` PrettyConst, Pretty fun) =>
             PrettyBy PrettyConfigPlc (CkValue uni fun) where
-    prettyBy cfg = prettyBy cfg . ckValueToTerm ()
+    prettyBy cfg = prettyBy cfg . ckValueToTerm
 
 data CkUserError =
     CkEvaluationFailure -- Error has been called or a builtin application has failed
     deriving (Show, Eq)
 
+-- | The CK machine-specific 'EvaluationException' parameterized over @term@.
+type CkEvaluationExceptionCarrying term fun =
+    EvaluationException CkUserError fun term
+
+-- See Note [Being generic over @term@ in 'CekM']
```

Comment:
The note is kinda big, so I don't want to replicate it in this file, given that it applies to the CK machine the same way it applies to the CEK one.